spring-xsuaa/README.md

```diff
@@ -145,6 +145,38 @@ In detail `com.sap.cloud.security.xsuaa.token.SpringSecurityContext` wraps the S
 
 Note that Spring Security Context is thread-bound and is NOT propagated to child-threads. This [Baeldung tutorial: Spring Security Context Propagation article](https://www.baeldung.com/spring-security-async-principal-propagation) provides more information on how to propagate the context.
 
+### Token Exchange
+In case application is required to do token exchange token-client and java-api dependencies need to be available(shouldn't be excluded) in the project.
+```xml
+<dependency>
+    <groupId>com.sap.cloud.security.xsuaa</groupId>
+    <artifactId>token-client</artifactId>
+</dependency>
+<dependency>
+    <groupId>com.sap.cloud.security</groupId>
+	<artifactId>java-api</artifactId>		
+</dependency>
+```
+To use the token exchange, in the application's security configuration `bearerTokenResolver` needs to be defined in the following manner:
+```java
+http.authorizeRequests()
+      .antMatchers("/secured/path/**").hasAuthority("application.Read")
+      .anyRequest().denyAll()
+      .and().oauth2ResourceServer()
+      .bearerTokenResolver(new TokenBrokerResolver(xsuaaServiceConfiguration, cache,
+                                 authenticationMethod));
+```
+where:
+- xsuaaServiceConfiguration: configuration properties from environment
+- cache: application's token-cache
```

Comment:
is cache optional? In any case further details might be required: code, link to token-client lib doc?

Reply:
doesn't seem to be optional and is used only within the TokenBrokerResolver class directly.

Reply:
I've checked the code and the constructors once more. So, the cache is only required in case of authentication method BASIC or CLIENT_CREDENTIALS... but not for pure OAuth... maybe we need a new/separate TokenBrokerResolverDefault implementation, that gets used within the existing one?

Reply:
if separate implementation of BearerTokenResolver class, then application developer needs to choose which TokenBrokerResolver to use in securityConfiguration, if that's fine and you think it doesn't complicate things then I can modify the existing IasXsuaaExchangeBroker class to implement the BearerTokenResolver, but that would not be seamless IAS Xsuaa exchange implementation, I guess?